docs/pai-job-protocol.yaml

```diff
@@ -0,0 +1,223 @@
+    version: String, optional # Component version, Default is latest.
+    contributor: String, optional
+    description: String, optional
+    auth: Object, optional # Only available when the type is dockerimage.
+      username: String, optional
+      passward: String, optional
+      registryuri: String, optional
+    uri: String or list, required # Only when the type is data can the uri be a list.
+
+# If specified, the whole parameters object can be referenced as `$parameters`.
+# Scope of reference `$parameters`: the refercen is global and shared among all task roles.
+parameters: # Optional, can be omitted.
+  <param1>: value1 # Specify name and value of all the referencable parameters that will be used in the whole job template.
+  <param2>: value2 # Can be referenced by `<% $parameters.param1 %>`, `<% $parameters.param2 %>`.
+
+jobRetryCount: Integer, optional # Default is 0.
+# Task roles are different types of task in the protocol.
+# One job may have one or more task roles, each task role has one or more instances, and each instance runs inside one container.
+taskRoles:
+  <name>: String, required  # Name of the taskRole, string in ^[A-Za-z0-9\-._~]+$ format.
+    instances: Integer, optional # Default is 1, instances of a taskRole, no less than 1.
+    completion: # Completion poclicy for the job, https://github.com/Microsoft/pai/blob/master/subprojects/frameworklauncher/yarn/doc/USERMANUAL.md#ApplicationCompletionPolicy.
+      # Number of failed tasks to fail the entire job, null or no less than 1, if set to null means the job will always succeed regardless any task failure.
+      minFailedInstances: Integer or null, optional # Default is 1.
+      # Number of succeeded tasks to succeed the entire job, null or no less than 1, if set to null means the job will only succeed until all tasks are completed and minFailedInstances is not triggered.
+      minSucceededInstances: Integer or null, optional # Default is null.
+    taskRetryCount: Integer, optional # Default is 0.
+    dockerImage: String, required # Should reference to a dockerimage defined in prerequisites.
+    # Scope of the reference `$data`, '$output', `$script`: the reference is only valid inside this task role.
+    # User cannot reference them from another task role. Reference for `$parameters` is global and shared among task roles.
+    data: String, optional # Should reference to data defined in prerequisites, the whole data object in this task role can be referenced as `$data`.
+    output: String, optional # Should reference to output defined in prerequisites, the whole output object in this task role can be referenced as `$output`.
+    script: String, optional # Should reference to script defined in prerequisites, the whole script object in this task role can be referenced as `$script`.
+    extraContainerOptions:
```

Comment:
And description for output and script? @yang.fan@163.com  what your opinion?

Reply:
Updated.